lib/actions/app.coffee

```diff
@@ -44,7 +45,13 @@ exports.create =
 					return callback(new Error('You already have an application with that name!'))
 
 				return callback(null, options.type) if options.type?
-				visuals.patterns.selectDeviceType(callback)
+				resin.models.device.getSupportedDeviceTypes().then (supportedDeviceTypes) ->
+					form.ask
+						message: 'Device Type'
+						name: 'deviceType'
+						type: 'list'
+						choices: supportedDeviceTypes
+					.nodeify(callback)
```

Comment:
In this case, if `resin.models.device.getSupportedDeviceTypes()` happens to throw an error, you will miss it, because `.nodeify()` only applies to `form.ask()`. You can move `.nodeify()` to match `resin.models.device.getSupportedDeviceTypes()` instead.